### PR TITLE
[auto_submit] Use SecretManager

### PR DESCRIPTION
**Pull request by CaseyHillers**
This is to support storing the secrets without Datastore.

# Test

I did some local runs and deploys to verify the APIs connected.

# Future work

Upstream this into `package:gcloud`

### CODE REVIEW

File: auto_submit/lib/service/secrets.dart [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+  Future<String> get(String key);
+
+  void put(String key, [String? value]);
+}
+
+class CloudSecretManager implements SecretManager {
+  CloudSecretManager();
+
+  final String projectId = Platform.environment['APPLICATION_ID'] ?? 'flutter-dashboard';
```
Comment (keyonghan) [inline auto_submit/lib/service/secrets.dart:25]: Nit: simply use `flutter-dashboard` as this is not local run and these secrets are in this project only.

**Review by keyonghan — APPROVED** [review]
LGTM with nit.

File: auto_submit/lib/service/secrets.dart (hunk 1 above)
Comment (CaseyHillers) [reply auto_submit/lib/service/secrets.dart:25]: I want to upstream this into `package:gcloud`, and I'm leaving this as a breadcrumb to sort out how to find this (googleapis surfaces it somewhere). SG?

**Review by keyonghan — APPROVED** [review]
LGTM

**Pull request merged**